### PR TITLE
allow for flow json to have incomplete deployments in flow json

### LINKED ISSUE #1202: Create account removes deployment section
If I have the following flow.json:

```
{
	"networks": {
		"emulator": "127.0.0.1:3569",
		"mainnet": "access.mainnet.nodes.onflow.org:9000",
		"sandboxnet": "access.sandboxnet.nodes.onflow.org:9000",
		"testnet": "access.devnet.nodes.onflow.org:9000"
	},
	"accounts": {
		"emulator-account": {
			"address": "f8d6e0586b0a20c7",
			"key": "09f3d3a54819c8004cb8b71169f0caf6f9df4528fce015ec0302797c15051d69"
		}
	},
	"deployments": {
		"emulator": {}
	}
}
```

and I go to create an account with `flow accounts create` and choose emulator it will remove my "deployments" section.

### PR DESCRIPTION
**Pull request by bthaile**
Closes #1202 

## Description

Allow flow.json to keep partial filled out deployments. Previously it would remove deployments if an account wasn't present.

______

For contributor use:

- [ ] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [ ] Code follows the [standards mentioned here](https://github.com/onflow/flow-cli/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

File: flowkit/config/json/deploy_test.go [hunk 1]
```diff
@@ -151,3 +151,33 @@ func Test_DeploymentAdvanced(t *testing.T) {
...
 	assert.Len(t, alice.Contracts[1].Args, 0)
 }
+
+func Test_EmptyEmulatorDeployment(t *testing.T) {
+	b := []byte(`{
+		"emulator": {}
+	}`)
+
+	result := []byte(`{
+		"emulator": {"":[]}
```
Comment (chasefleming) [inline flowkit/config/json/deploy_test.go:161]: Seems odd to replace it with this. Is there no way to keep it as it was?

Comment (bthaile) [reply flowkit/config/json/deploy_test.go]: I added a else if to check if only network is set in a deployment and allow for a empty deployments object

**Review by chasefleming — CHANGES_REQUESTED** [review]
Tests pass, but it raises a new issue when I run it. I get the error:

`❌ Config Error: deployment contains nonexisting account `

Apparently it now things an empty object contains an account somewhere in the `flow accounts create` flow.

**Comment by bthaile** [comment]
Closing, we need to verify the dev process of running cli commands to make sure the flow.json stays correct.

**Pull request closed without merging**